### ISSUE TITLE
Problems with assign behavior

### ISSUE THREAD

**Issue by ryanbadman** [joshcullen/bayesmove#9]
Hello, I am just trying to replicate the demo and getting a strange error at the very end when I try to assign behavior to the original data file. Can you assist? I attached the data file I was using and the R code I was running. Everything works up to that point:

tracks_processed.out<- assign_behavior(dat.orig = tracks0,
                             dat.seg.list =  df_to_list(tracks_processed.seg, "id"),
                             theta.estim.long = theta.estim.long,
                             behav.names = NULL)

# Merge results with original data
> tracks_processed.out<- assign_behavior(dat.orig = tracks0,
+                              dat.seg.list =  df_to_list(tracks_processed.seg, "id"),
+                              theta.estim.long = theta.estim.long,
+                              behav.names = NULL)
Error in `dplyr::mutate()`:
ℹ In argument: `behav = behav.names1[apply(sub[, 5:ncol(sub)], 1, which.max)]`.
Caused by error in `FUN()`:
! 'list' object cannot be coerced to type 'double'
Run `rlang::last_trace()` to see where the error occurred.
Warning message:
Values from `prop` are not uniquely identified; output will contain list-cols.
• Use `values_fn = list` to suppress this warning.
• Use `values_fn = {summary_fun}` to summarise duplicates.
• Use the following dplyr code to identify duplicates.
  {data} |>
  dplyr::summarise(n = dplyr::n(), .by = c(id, tseg, time1, date, behavior)) |>
  dplyr::filter(n > 1L) 
  
[scalars_alive_cat100_final.csv](https://github.com/joshcullen/bayesmove/files/15382875/scalars_alive_cat100_final.csv)

**Comment by ryanbadman** [comment]
Here is the R file: 
[RFileText.txt](https://github.com/joshcullen/bayesmove/files/15383104/RFileText.txt)

**Comment by joshcullen** [comment]
That seems similar to the bug I had fixed for the most recent version of `bayesmove` available on GitHub (v0.2.3). The version on CRAN is v0.2.1 I believe. Do you know which version you're using? Also, you should add a vector of behavior names to the `behav.names` argument, which is likely the cause of the first error since it's left NULL.

I actually ran into a problem when running the segmentation model via `segment_behavior()` function on Line 194 of your R script. This looks to be related to the bin limits for turning angles compared to the underlying data. When using the even bin limits for the turning angle (i.e., pi/4), each of the IDs only contain some, but not all, of the 8 possible bins. The model requires that all bins have at least 1 but ideally more observations per bin. So this comes down to redefining the bin limits to better characterize your dataset: in this particular case, adding more bin to break up the small angles close to 0 radians and likely having fewer (but larger) bins representing angles close to -pi or +pi due to fewer data points in this range. This may take a little trial and error as you discretize and then visualize the turning angles compared to these limits.

**Comment by ryanbadman** [comment]
Thanks for the quick reply! I used CRAN and got the first error that started this post, I switched to the stable Github version yesterday also (sorry should have specified) and got the same error even when I made the behav.names null which I did because it threw a new error on the github version when I didn't:

Error in assign_behavior(dat.orig = tracks0, dat.seg.list = df_to_list(tracks_processed.seg,  : 
  The 'behav.names' argument has been deprecated. Don't specify this argument
  
  I have reattached an R version with the angle bins broken up in a smarter way. FYI I was just running this script on some lower quality test data that is formatted a certain way to make sure the full pipeline works, so that's why I wasn't being too careful with bin stuff yet. But agree it may help debugging especially if you got some flag from my bad first bins. 
  
And actually it worked! It seems maybe it was some fit quality issue that caused downstream issues with very vague errors, now the full pipeline works. Thanks!
[RFileText_v2.txt](https://github.com/joshcullen/bayesmove/files/15391904/RFileText_v2.txt)

Here is the final script if it helps.
